File: LowestTerm.py
# A class
class LowTerm:

    # A constructor
    def __init__(self, fraction):
        self.numerator   = fraction['numerator']
        self.denominator = fraction['denominator']

        # set trial_factor to the smaller value between numerator and denominator
        if self.numerator < self.denominator: self.trial_factor = self.numerator
        else: self.trial_factor = self.denominator

    # Returns a dictionary of the new numerator and denominator
    def reduceFraction(self):
        # We are counting down to test for mutual factors 
        while self.trial_factor > 1:
            # do we have a factor
            if self.numerator % self.trial_factor == 0:
                # is this factor mutual?
                if self.denominator % self.trial_factor == 0:
                    # where we have a mutual factor
                    self.numerator   /= self.trial_factor
                    self.denominator /= self.trial_factor
                    continue

            self.trial_factor -= 1

        return {'numerator':int(self.numerator), 'denominator':int(self.denominator)}

File: LowestTerm.vb
Public Class LowestTerm

    Dim numerator As Integer
    Dim denominator As Integer
    Dim trial_factor As Integer

    ' A constructor
    Public Sub _init_(fraction As Dictionary(Of String, Integer))
        numerator = fraction.Item("numerator")
        denominator = fraction.Item("denominator")

        ' Set trial_factor To the smaller value between numerator And denominator
        If numerator < denominator Then
            trial_factor = numerator
        Else
            trial_factor = denominator
        End If
    End Sub

    ' Returns a dictionary Of the New numerator And denominator
    Public Function reduceFraction() As Dictionary(Of String, Integer)
        ' We are counting down To test For mutual factors 
        Do While trial_factor > 1
            ' Do we have a factor
            If numerator Mod trial_factor = 0 Then
                ' Is this factor mutual?
                If denominator Mod trial_factor = 0 Then
                    ' where we have a mutual factor
                    numerator /= trial_factor
                    denominator /= trial_factor
                    Continue Do
                End If
            End If

            trial_factor -= 1
        Loop

        Dim send_back As New Dictionary(Of String, Integer)
        send_back.Add("numerator", numerator)
        send_back.Add("denominator", denominator)
        Return send_back

    End Function

End Class
